maploc/data/dataset.py

```diff
@@ -126,63 +126,74 @@ def get_view(self, idx, scene, seq, name, seed, bbox_tile):
         cam_dict = self.data["cameras"][scene][seq][self.data["camera_id"][idx]]
         cam = Camera.from_dict(cam_dict).float()
 
+        # for backward compatibility
         if "roll_pitch_yaw" in self.data:
-            roll, pitch, yaw = self.data["roll_pitch_yaw"][idx].numpy()
+            world_R_cam = compose_rotmat(*self.data["roll_pitch_yaw"][idx].numpy())
         else:
-            roll, pitch, yaw = decompose_rotmat(self.data["R_c2w"][idx].numpy())
+            world_R_cam = self.data["R_c2w"][idx].numpy()
+
+        world_t_cam = self.data["t_c2w"][idx].numpy()
+
         image = read_image(self.image_dirs[scene] / (name + self.image_ext))
 
-        if "plane_params" in self.data:
-            # transform the plane parameters from world to camera frames
-            plane_w = self.data["plane_params"][idx]
-            data["ground_plane"] = torch.cat(
-                [rotmat2d(deg2rad(torch.tensor(yaw))) @ plane_w[:2], plane_w[2:]]
-            )
         if self.cfg.force_camera_height is not None:
             data["camera_height"] = torch.tensor(self.cfg.force_camera_height)
         elif "camera_height" in self.data:
             data["camera_height"] = self.data["height"][idx].clone()
 
         # raster extraction
         canvas = self.tile_managers[scene].query(bbox_tile)
-        xy_w_gt = self.data["t_c2w"][idx][:2].numpy()
-        uv_gt = canvas.to_uv(xy_w_gt)
-        uv_init = canvas.to_uv(bbox_tile.center)
         raster = canvas.raster  # C, H, W
+        raster = torch.from_numpy(np.ascontiguousarray(raster)).long()
+        world_T_cam = Transform3D.from_Rt(world_R_cam, world_t_cam)
+        world_T_cam2d = Transform2D.camera_2d_from_3d(world_T_cam)
+
+        _, cam_R_gcam = decompose_cam_into_gcam(world_T_cam)
+
+        world_T_tile = Transform2D.from_Rt(torch.eye(2), canvas.bbox.min_)
+        tile_T_cam = (world_T_tile.inv() @ world_T_cam2d).float()
+
+        image, valid, cam = self.process_image(image, cam, seed, cam_R_gcam)
 
         # Map augmentations
-        heading = np.deg2rad(90 - yaw)  # fixme
         if self.stage == "train":
             if self.cfg.augmentation.rot90:
-                raster, uv_gt, heading = random_rot90(raster, uv_gt, heading, seed)
+                raster, tile_T_cam = random_rot90(raster, tile_T_cam, canvas.ppm)
             if self.cfg.augmentation.flip:
-                image, raster, uv_gt, heading = random_flip(
-                    image, raster, uv_gt, heading, seed
+                image, valid, raster, tile_T_cam = random_flip(
+                    image, valid, raster, tile_T_cam, canvas.ppm
```

Comment:
Should we set set `cam_R_gcam = None` in this case to make sure that it's not used subsequently?

Reply:
I'm pushing an update soon with the correct value for cam_R_gcam, in case we need to use it for anything later

Reply:
Pushed here 9580558. The rectification now happens with the updated `cam_R_gcam` after the augmentations. If we don't really need the correct `cam_R_gcam` later, we could maybe undo this change and do what you suggested - set `cam_R_gcam = None`.